### DISCUSSION TITLE
NamedRows::next documentation

### DISCUSSION THREAD

**Discussion by dcz-self** [cozodb/cozo#136] [Ideas]
Originally I thought NamedRows::next was a kind of a cursor structure, to avoid storing large results all at once in memory, but after taking a closer look, the chain stores all data together anyway.

The instance in `next` can also store its own columns. Is this for reporting results from multiple queries?

Could the documentation explain when the user can expect `next` to be populated, and whether the next set of rows is actually independent from the first?

**Comment by zh217** [comment]
It is only used for returning multiple relations within an imperative script, which may also be undocumented (I can't remember).

I will remind myself to document these when the next release comes.

**Comment by dcz-self** [comment]
I mention this because if the columns can change between instances, then my latest pull request is buggy.

**Reply by zh217** [reply]
Thank you for pointing this out. I didn't look at that carefully enough. Can you make an update pull request?

**Reply by dcz-self** [reply]
If columns can be different, then the entire change must be reverted. I don't know how to improve that API without knowing more about the `next` field.

**Reply by zh217** [reply]
I see. I will revert it.
